src/workers/parser/mapExpression.js

```diff
@@ -7,6 +7,22 @@
 import mapOriginalExpression from "./mapOriginalExpression";
 import mapExpressionBindings from "./mapBindings";
 
+import { hasSyntaxError } from "./validate";
+import { buildScopeList } from "./getScopes";
+import generate from "@babel/generator";
+import * as t from "@babel/types";
+
+function handleTopLevelAwait(expression) {
+  if (hasSyntaxError(expression) && expression.match(/await/)) {
```

Comment:
we should probably be smarter here :)

Reply:
I'd re-parse with https://github.com/babel/babel/blob/master/packages/babel-parser/src/options.js#L32 enabled and then use an AST transform to do this wrapping.

Reply:
makes sense!